### ISSUE TITLE
REST API - No API provided for Login

### ISSUE THREAD

**Issue by Sebastian-Schuchmann** [buddyboss/buddyboss-platform#3147]
Most API's listed in your REST API Documentation require the permission "LoggedInUser", which of course makes sense. But I can't find a way to login via the API itself. There are only two Permission listed under "WithoutLoggedInUser", which are Signup Form and Create singup but both seem to be for creating a new account and not logging into an existing one. 

Is there something I am missing. We are trying to access the Platform running on our Wordpress site via an App we are developing right now and just need the ability to create a new activity. 

Is this not what the API is intended for, am I misunderstanding something?

**Comment by meetmakadiya** [comment]
@Sebastian-Schuchmann 
There is a jwt token you need to generate and that token is used in API for login so there is an API for that token below.
`SiteURL/wp-json/buddyboss-app/auth/v1/jwt/token`
above API you need to pass the username, password and app_id  then you receive one token. http://i.imgur.com/gtHwmjo.png
once you get the token then you need to pass that token as an accessToken in the header for API. http://i.imgur.com/hkeFgi7.png

**Comment by BoweFrankema** [comment]
I am running into the same issue, where I can't figure out how to login as a user via the rest API in the non-app version of the BuddyBoss API.

You say this is not needed, you just need to use Basic Authentication (akin to the WP Rest API) and it should work. When I use basic auth I can use the WP JSON v2 API fully to create posts etc etc. But when I try to hit any BuddyBoss Platform endpoint like.

install + "/wp-json/buddyboss/v1/activity"

I get

{
  "code": "bp_rest_authorization_required",
  "message": "Sorry, you are not allowed to create activities.",
  "data": {
    "status": 403
  }
}

I've tried various things over the last hours, but I am not getting anywhere.. It would be awesome if your documentation could provide a simple outline on how to get authenticated/logged in for the BuddyBoss Platform (Web version). The rest of the documentation seems excellent :-)

@Sebastian-Schuchmann where you able to figure this out?

**Comment by BoweFrankema** [comment]
I've had some sleep and review the code of the Rest API and realised the only check was the is_user_logged_in function. So it had to be a problem in the methods I used via the rest API. So I followed the following tutorial which still works with getting the BuddyBoss REST API to work. 

https://apppresser.com/buddypress-rest-api/

I do think it would make sense to add the initial setup steps to your documentation because I believe it would help others!
